src/client/tls.rs

```diff
@@ -10,45 +10,32 @@ use futures::ready;
 use std::cmp;
 use std::{
     io,
-    mem::MaybeUninit,
     pin::Pin,
     task::{self, Poll},
 };
-use tokio::{
-    io::{AsyncRead, AsyncWrite},
-    net::TcpStream,
-};
 #[cfg(feature = "tls")]
-use tokio_tls::TlsStream;
+use async_native_tls::TlsStream;
 #[cfg(feature = "tls")]
 use tracing::{event, Level};
 
 /// A wrapper to handle either TLS or bare connections.
-pub(crate) enum MaybeTlsStream {
-    Raw(TcpStream),
+pub(crate) enum MaybeTlsStream<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.